Add command system

dreadnoughtsix opened the pull request:
The bot can now take commands with the exclamation point (!) as the prefix.

Commit:
Remove client parameter in run

The client parameter was used incorrectly, so changing the bot's status
when a specific condition is reached was not working properly. Need to
check on how to change the bot (logout, change status, etc.) when
necessary.

Commit:
Add basic commands

Commit:
Merge branch 'wip/add-arguments' into wip/add-commands

Add the preliminary argument cases into wip/add-commands for a better
tracking of most of the work added.

tpenguinltg reviewed (changes requested):
Many of the comments are for using `let` where `const` is more appropriate. Use `const` if you don't need to modify the value of the variable (in the case of a primitive) or the reference to the variable (in the case of an object or array). The other comments address refactoring opportunities.

Commit:
Add interval variable

The interval variable will be changed when the user inputs a valid food
item as Stubbs will have to sniff it first before eating it.

Commit:
Move Rock Paper Scissors into a module

It makes it so that it can be easier to add more games later on. The
modules are accessed using require(...).

Commit:
Change play group to interact group

Having a play command, it does not make sense for it to have a play
group as well. Renaming it to interact would make sense as the user
interacts with Stubbs using these commands.

Commit:
Rename play to interact

Commit:
Fix call for RPSGame

Commit:
Refactor CoinFlip game into module

Instead of having the CoinFlip game to be a part of the play module, it
will be separate and added into the games directory as an accessible
game.

Commit:
Change food limit to 5

Commit:
Add valid foods for restrictions

Commit:
Make food and max food count configurable

The list of valid foods and the max food count can now be configurable
using the separate config file.

Commit:
Refactor diceroll into play
